aerogear/mobile-security-service-operator
# feat(Add specific watch namespaces) : Impl workaround in Reconcile (AEROGEAR-9089)

## Motivation
https://issues.jboss.org/browse/AEROGEAR-9089

## What
* Add Env Var to specify the namespaces which could be used to apply the APP CR
* Add workaround to not allow the CRs to be applied in namespaces which are not the specific ones until the PR https://github.com/operator-framework/operator-sdk/pull/1388 be merged and we are able to attend this need properly. 
* Update readme

## Why
* Do not allow the operator apply the APP CR which is applied in a not specified namespace
* Do not allow the mss CR and its database be applied in a namespace which is not the same of the operator. 

## How
* Doing a check on reconciling to not allow it works in a namespace that is not one of the lists defined in the ENV VAR. 
* Doing the same above check in the reconciling of the Service and DB CRs but by checking if is the operator namespace or not. 

## Verification Steps
* Check the README:https://github.com/aerogear/mobile-security-service-operator/blob/d649ede996a013ea323a182e3901c8956ab1a7ac/README.adoc
* Use this PR and the image `docker.io/cmacedo/mobile-security-service-operator:AEROGEAR-9089`
* Install the operator by running "make create-all"
* Check that all were installed with success
* Create the default app namespace "oc new-project mobile-security-service-apps"
* Run make create-app
* Check that the configMap was created in this namespace for this app
* Create another namespace with any other name
* Run `make create-all` inside of this new namespace
* Ensure that the configMap was NOT created. 



## Checklist:


## Progress

- [] TODO
